src/lib/util/anchor.ts

```diff
@@ -22,7 +20,9 @@ export function ensureAnchorHistory () {
 		// We want to allow links to absolute paths without
 		// using the history API. Also, if the target is not this frame we
 		// allow the default behavior (eg. if it should open in a new tab)
-		if (!href.startsWith(location.origin) || ($anchor.target !== "" && $anchor.target !== "_self")) {
+		if (!href.startsWith(location.origin) ||
```

Comment:
@andreasbm according to the comment above, it seems this is also meant to allow links with absolute paths to work as-is and not use History api. However, the href property of the anchor always includes the origin. So this only checks if links are navigating _away_ from the site.

So as is, there is no way to have a link to another part of your site and _not_ use the History api. You would either need a way to opt out of it, like this PR is adding, or do an additional check like:
```js
$anchor.getAttribute("href").startsWith(location.origin) // checks for absolute urls to the current site.
```